R/dbi-result.R

```diff
@@ -126,17 +126,54 @@ setMethod("dbGetRowsAffected", "OdbcResult",
   }
 )
 
+#' @param param_description A data.frame with per-parameter attribute
+#' overrides.  Argument is optional; if used it must have columns:
+#' * param_index Index of parameter in query ( beginning with 1 ).
+#' * data_type Integer corresponding to the parameter SQL Data Type.
+#'   See \code{\link{ODBC_TYPE}}.
+#' * column_size Size of parameter.
+#' * decimal_digits Either precision or the scale of the parameter
+#'   depending on type.
+#'
+#' For more information see the [ODBC SQLBindParam documentation](https://learn.microsoft.com/en-us/sql/odbc/reference/syntax/sqlbindparameter-function?view=sql-server-ver16)
+#'
+#' @examples
+#' \dontrun{
+#' library(odbc)
+#' # Writing UNICODE into a VARCHAR
+#' # column with SQL server
+#' DBI::dbRemoveTable(conn, "#tmp")
+#' dbExecute(conn, "CREATE TABLE #tmp (col1 VARCHAR(50) COLLATE Latin1_General_100_CI_AI_SC_UTF8);")
+#' res <- dbSendQuery(conn, "INSERT INTO #tmp SELECT ? COLLATE Latin1_General_100_CI_AI_SC_UTF8")
+#' description <- data.frame(param_index = 1, data_type = ODBC_TYPE("WVARCHAR"),
+#'   column_size = 5, decimal_digits = NA_integer_)
+#' dbBind(res, params = list("\u2915"), param_description = description)
+#' dbClearResult(res)
+#' DBI::dbReadTable(conn, "#tmp")
+#' }
```

Comment:
```suggestion
#' @examplesIf FALSE
#' # Writing UNICODE into a VARCHAR column with SQL server
#' dbRemoveTable(conn, "#tmp")
#' 
#' dbExecute(
#'   conn, 
#'   "CREATE TABLE #tmp (col1 VARCHAR(50) COLLATE Latin1_General_100_CI_AI_SC_UTF8);"
#' )
#' 
#' res <- dbSendQuery(
#'   conn, 
#'   "INSERT INTO #tmp SELECT ? COLLATE Latin1_General_100_CI_AI_SC_UTF8"
#' )
#' 
#' description <- data.frame(
#'   param_index = 1, 
#'   data_type = ODBC_TYPE("WVARCHAR"),
#'   column_size = 5, 
#'   decimal_digits = NA_integer_
#' )
#' 
#' dbBind(res, params = list("\u2915"), param_description = description)
#' dbClearResult(res)
#' 
#' dbReadTable(conn, "#tmp")
```

Some style edits I'd make:

* `examples` with `\dontrun{}` -> `examplesIf FALSE` so that users don't see `## Not run` tags and CRAN definitely won't try to run the examples
* I _believe_ the `DBI::` namespacing here may not be needed?
* Line-breaking more liberally via [codegrip](https://github.com/lionel-/codegrip)